### ISSUE TITLE
`appimagetool` messages contradict: First, "AppStream upstream metadata found in" then second:  "No AppStream metadata was found" 

### ISSUE THREAD

**Issue by KurtPfeifle** [AppImage/AppImageKit#634]
Consider the following Travis log from an ImageMagick build:

https://travis-ci.org/KurtPfeifle/ImageMagick/builds/333429134

Lines 4753-4762 read:

```
Using architecture x86_64
App name for filename: ImageMagick
dest_path: ImageMagick-x86_64.AppImage
AppStream upstream metadata found in usr/share/metainfo/imagemagick.appdata.xml
Desktop file: /home/travis/build/KurtPfeifle/ImageMagick/appdir/imagemagick.desktop
/home/travis/build/KurtPfeifle/ImageMagick/appdir should be packaged as ImageMagick-git.eb92949-x86_64.AppImage
Trying to validate AppStream information with the appstreamcli tool
In case of issues, please refer to https://github.com/ximion/appstream
I - <unknown>:<root>
    No AppStream metadata was found.
```

Why, oh why? First, in line 4756 *"AppStream upstream metadata found"*, then 6 lines further down: *"No AppStream metadata was found"*.

Also, see lines 3071-3077:

```
$ export _appstreamcli="$(appstreamcli validate --verbose ./appimage/imagemagick.appdata.xml 2>&1)"
The command "export _appstreamcli="$(appstreamcli validate --verbose ./appimage/imagemagick.appdata.xml 2>&1)"" exited with 0.
$ echo "$_appstreamcli"
Validation was successful.
```

**Comment by probonopd** [comment]
> First, in line 4756 "AppStream upstream metadata found"

This is `linuxdeployqt` saying "I found a `usr/share/metainfo/*.appdata.xml` file

> then 6 lines further down: "No AppStream metadata was found"

This is `appstreamcli` saying "<unknown>:<root> is somehow wrong and I don't consider this file as valid AppStream metadata". I agree that maybe this could be worded more clearly; cc @ximion.

> Also, see lines 3071-3077

I am clueless...

**Comment by ximion** [comment]
@probonopd No, I think something else is going on here... This message of `appstreamcli` basically says that the metainfo directory doesn't exist - otherwise it would have given a reason why the found metadata file was invalid.
So, I would check which path gets passed to `appstreamcli validate-tree` here.

**Comment by TheAssassin** [comment]
Part of #603.

**Comment by ximion** [comment]
This is a bug in whatever script called appstreamcli, and not the tool itself.

**Comment by probonopd** [comment]
Closing since all involved codebases since have evolved a lot. Feel free to reopen if the issue persists. Thanks for reporting.
